### PR TITLE
Emit ResourceWarning on unclosed memory object streams

### PR DESCRIPTION
**Pull request by agronholm**
This resurrects #663, but in a simpler format.

### CODE REVIEW

File: src/anyio/_backends/_asyncio.py [hunk 1]
```diff
@@ -1922,6 +1920,8 @@ async def _call_in_runner_task(
                 self._run_tests_and_fixtures(receive_stream)
             )
 
+            self._runner_task.add_done_callback(lambda _: self._send_stream.close())
```
Comment (graingert) [inline src/anyio/_backends/_asyncio.py:1923]: it might be nicer to do:

```
async def coro():
    try:
        self._run_tests_and_fixtures(receive_stream)
     finally:
         self._send_stream.close()

self._runner_task = self.get_loop().create_task(coro())
```
?

**Review by graingert — APPROVED** [review]

File: src/anyio/_backends/_asyncio.py [hunk 2]
```diff
@@ -1838,8 +1838,6 @@ def _create_task_info(task: asyncio.Task) -> TaskInfo:
 
 
 class TestRunner(abc.TestRunner):
-    _send_stream: MemoryObjectSendStream[tuple[Awaitable[Any], asyncio.Future[Any]]]
```
Comment (graingert) [inline src/anyio/_backends/_asyncio.py:1841]: why did you remove this?

Comment (agronholm) [reply src/anyio/_backends/_asyncio.py:1841]: I didn't write this code, I merged from the contributor's branch.

Comment (agronholm) [reply src/anyio/_backends/_asyncio.py]: I've just reverted that change.

File: src/anyio/_backends/_asyncio.py (hunk 1 above)
Comment (agronholm) [reply src/anyio/_backends/_asyncio.py]: How about now?

**Review by graingert — APPROVED** [review]
